wallet/wallet.go

```diff
@@ -2512,9 +2512,19 @@ func (w *Wallet) ListAddressTransactions(ctx context.Context, pkHashes map[strin
 
 				for _, cred := range detail.Credits {
 					pkScript := detail.MsgTx.TxOut[cred.Index].PkScript
-					_, addrs, _, err := txscript.ExtractPkScriptAddrs(
-						0, pkScript, w.chainParams, true) // Yes treasury
-					if err != nil || len(addrs) != 1 {
+					version := detail.MsgTx.TxOut[cred.Index].Version
+
+					// Instead of ExtractAddrs -> Hash160, just extract pkh?
```

Comment:
This is what I would suggest.  It's both simpler and significantly more efficient since it avoids all the extra address creation and associated allocations.

Reply:
Agree, changing.  I'm not confident that there's test coverage for this method, so I just did a quick check on the playground: https://go.dev/play/p/tF58hb1UMPP
Sorry if that's cringe, but I wanted to demo.